include/libembeddedhal/spi/spi.hpp

```diff
@@ -72,6 +76,6 @@ class spi
   virtual boost::leaf::result<void> driver_transfer(
     std::span<const std::byte> p_data_out,
     std::span<std::byte> p_data_in,
-    std::byte p_filler) noexcept = 0;
+    std::byte p_filler = default_filler) noexcept = 0;
```

Comment:
```suggestion
    std::byte p_filler) noexcept = 0;
```

This actually goes against [core guidelines C.140](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c140-do-not-provide-different-default-arguments-for-a-virtual-function-and-an-overrider)

Everywhere else is fine.